client/pages/post/[id].tsx

```diff
@@ -6,13 +6,16 @@ import { useEffect, useState } from "react";
 import Document from '../../components/document'
 import Header from "../../components/header";
 import VisibilityBadge from "../../components/visibility-badge";
-import { ThemeProps } from "../_app";
+import { PostProps } from "../_app";
 import PageSeo from "components/page-seo";
 import Head from "next/head";
 import styles from './styles.module.css';
 import Cookies from "js-cookie";
+import cookie from "cookie";
+import { GetServerSideProps } from "next";
 
-const Post = ({ theme, changeTheme }: ThemeProps) => {
+
+const Post = ({renderedPost, theme, changeTheme}: PostProps) => {
     const [post, setPost] = useState<any>()
```

Comment:
Can you make this `useState<PostProps>(renderedPost)` and refactor the useEffect logic to reflect the default being set here?

Reply:
renderPost doesn't have a type associated with it. 

So what I will do is just `useState(renderedPost)`. Code is cleaned up like that and if down the road, a type for Post is created, it can be added.

The reason I am not creating a type for Post myself is I'm not too familiar with the JSON structure of a post. You're probably the best person to do that.

Reply:
Also noticed I needed to do some proper error handling. Added a try block around the post.json()